x-pack/filebeat/tests/integration/status_reporter_test.go

```diff
@@ -0,0 +1,217 @@
+
+	"github.com/elastic/elastic-agent-client/v7/pkg/client"
+	"github.com/elastic/elastic-agent-client/v7/pkg/client/mock"
+	"github.com/elastic/elastic-agent-client/v7/pkg/proto"
+
+	"github.com/elastic/beats/v7/libbeat/common/reload"
+	lbmanagement "github.com/elastic/beats/v7/libbeat/management"
+	"github.com/elastic/beats/v7/x-pack/filebeat/cmd"
+	"github.com/elastic/beats/v7/x-pack/libbeat/management"
+	"github.com/elastic/beats/v7/x-pack/libbeat/management/tests"
+
+	conf "github.com/elastic/elastic-agent-libs/config"
+)
+
+func TestLogStatusReporter(t *testing.T) {
+	unitOneID := mock.NewID()
+	unitOutID := mock.NewID()
+	token := mock.NewID()
+
+	tests.InitBeatsForTest(t, cmd.Filebeat())
+	tmpDir := t.TempDir()
+	filename := fmt.Sprintf("test-%d", time.Now().Unix())
+	outPath := filepath.Join(tmpDir, filename)
+	t.Logf("writing output to file %s", outPath)
+	err := os.Mkdir(outPath, 0775)
+	require.NoError(t, err)
+	defer func() {
+		err := os.RemoveAll(outPath)
+		require.NoError(t, err)
+	}()
+
+	/*
+	 * valid input stream, shouldn't raise any error.
+	 */
+	inputStream := getInputStream(unitOneID, filepath.Join(tmpDir, "*"), 2)
+
+	/*
+	 * try to open an irregular file.
+	 * This should throw "Tried to open non regular file:" and status to degraded
+	 */
+	nullDeviceFile := "/dev/null"
+	if runtime.GOOS == "windows" {
+		nullDeviceFile = "NUL"
+	}
+	inputStreamIrregular := getInputStream(unitOneID, nullDeviceFile, 1)
+
+	outputExpectedStream := proto.UnitExpected{
+		Id:             unitOutID,
+		Type:           proto.UnitType_OUTPUT,
+		ConfigStateIdx: 1,
+		State:          proto.State_HEALTHY,
+		Config: &proto.UnitExpectedConfig{
+			DataStream: &proto.DataStream{
+				Namespace: "default",
+			},
+			Type:     "file",
+			Revision: 1,
+			Meta: &proto.Meta{
+				Package: &proto.Package{
+					Name:    "system",
+					Version: "1.17.0",
+				},
+			},
+			Source: tests.RequireNewStruct(map[string]interface{}{
+				"type":            "file",
+				"enabled":         true,
+				"path":            outPath,
+				"filename":        "beat-out",
+				"number_of_files": 7,
+			}),
+		},
+	}
+
+	observedStates := make(chan *proto.CheckinObserved)
+	expectedUnits := make(chan []*proto.UnitExpected)
+	done := make(chan struct{})
+	// V2 mock server
+	server := &mock.StubServerV2{
+		CheckinV2Impl: func(observed *proto.CheckinObserved) *proto.CheckinExpected {
+			select {
+			case observedStates <- observed:
+				return &proto.CheckinExpected{
+					Units: <-expectedUnits,
+				}
+			case <-done:
+				return nil
+			}
+		},
+		ActionImpl: func(response *proto.ActionResponse) error {
+			return nil
+		},
+	}
+	require.NoError(t, server.Start())
+	defer server.Stop()
+
+	// start the client
+	client := client.NewV2(fmt.Sprintf(":%d", server.Port), token, client.VersionInfo{
+		Name: "program",
+		Meta: map[string]string{
+			"key": "value",
+		},
+	}, client.WithGRPCDialOptions(grpc.WithTransportCredentials(insecure.NewCredentials())))
```

Comment:
Same thing here, optional fields can be omitted:
```suggestion
	client := client.NewV2(fmt.Sprintf(":%d", server.Port), token, client.VersionInfo{
		Name: "program",
	}, client.WithGRPCDialOptions(grpc.WithTransportCredentials(insecure.NewCredentials())))
```